examples/bin/sample_internet_client.rs

```diff
@@ -0,0 +1,55 @@
+use rustls_symcrypt::default_symcrypt_provider;
+
+use std::io::{stdout, Read, Write};
+use std::net::TcpStream;
+use std::sync::Arc;
+
+/// Usage
+/// This program provides a simple internet client using rustls-symcrypt to connect to rust-lang.org
+/// It uses the provided RootCA.pem to set up the client config.
+/// To run this program you can use the following command in your terminal:
+/// cargo run --bin sample_internet_client
+fn main() {
+    let root_store = rustls::RootCertStore {
+        roots: webpki_roots::TLS_SERVER_ROOTS.iter().cloned().collect(),
+    };
+
+    let config = rustls::ClientConfig::builder_with_provider(Arc::new(default_symcrypt_provider()))
+        .with_safe_default_protocol_versions()
+        .unwrap()
+        .with_root_certificates(root_store)
+        .with_no_client_auth();
+
+    let server_name = "www.rust-lang.org".try_into().unwrap();
+    let mut sock = TcpStream::connect("www.rust-lang.org:443").unwrap();
```

Comment:
>"www.rust-lang.org:443" [](http://example.com/codeflow?start=38&length=23)

Might be overkill, but you could specify the server and port as constants so that users can easily change them for their own testing.

Reply:
IMO a bit overkill, this is boilerplate code copied from rustls. On their examples they have a much more detailed client that has that sort of configuration abilities.